python/graphstorm/gpartition/local_partition_graph.py

```diff
@@ -0,0 +1,97 @@
+"""
```

Comment:
Shall we rename it to partition_algo.py?
The name local_partition_graph.py is a little bit confusing as we also have dist_partition_graph.py but with totally different impl.

Reply:
Good point, I'll rename to `partition_algo_base.py`